src/main/java/org/opensearch/ad/ml/ModelManager.java

```diff
@@ -817,20 +819,22 @@ public ThresholdingResult getAnomalyResultForEntity(
                 modelState.setModel(entityModel);
             }
 
-            // trainModelFromExistingSamples may be able to make models not null
-            if (entityModel.getRcf() == null || entityModel.getThreshold() == null) {
-                entityColdStarter.trainModelFromExistingSamples(modelState);
-            }
-
-            if (entityModel.getRcf() != null && entityModel.getThreshold() != null) {
-                return score(datapoint, modelId, modelState);
+            if (entityModel.getErcf().isPresent()) {
+                result = toResult(entityModel.getErcf().get().process(datapoint));
             } else {
-                entityModel.addSample(datapoint);
-                return new ThresholdingResult(0, 0, 0);
+                // trainModelFromExistingSamples may be able to make models not null
+                if (entityModel.getRcf() == null || entityModel.getThreshold() == null) {
```

Comment:
Should we remove such RCF and Threshold things as ERCF hide these behind the scene ?

Reply:
Yes, eventually. This branch dealing with existing models will be replaced with new pr.